leaf-ui/scripts/main.js

```diff
@@ -1034,54 +1034,57 @@ graph.on('change:size', function(cell, size){
 //Removing a link
 this.graph.on('remove', function(cell, collection, opt) {
    if (cell.isLink()) {
-	   //Verify if is a Not both type. If it is remove labels from source and target node
-	   var link = cell;
-	   var source = link.prop("source");
-	   var target = link.prop("target");
-
-	   for(var i = 0; i < graph.getElements().length; i++ ){
-		   if(graph.getElements()[i].prop("id") == source["id"]){
-			   source = graph.getElements()[i];
+	   if(cell.prop("link-type") == 'NBT' || cell.prop("link-type") == 'NBD'){
```

Comment:
This is the only line changed.
I wasn't verifying if the link deleted was NBT or NBD so it was removing all functions labels for all types of functions.